bugfix: S3C-4553 fix handling of special characters

jonathan-gramain opened the pull request:
Fix possible infinite loop during listing because of missing encoding
of special characters when requesting bucketd, in particular, the
ampersand.

jonathan-gramain committed:
bugfix: S3C-4553 fix handling of special characters

Fix possible infinite loop during listing because of missing encoding
of special characters when requesting bucketd, in particular, the
ampersand.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/s3utils/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

jonathan-gramain commented:
/approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/s3utils/commit/fad5e6b086d8aa907548584092c7340e070e99f1) did not succeed in branch bugfix/S3C-4553-fixHandlingOfSpecialCharacters.




*The following options are set:* **approve**

jonathan-gramain committed:
update eslint dependency

Set commit hash from Guidelines instead of default 'master' branch
which is unmaintained. This in turn allows lines of up to 120
characters instead of 80.

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.9`



The following branches will **NOT** be impacted:

* `development/1.8`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.9`



The following branches have **NOT** changed:

* `development/1.8`



Please check the status of the associated issue S3C-4553.

Goodbye jonathan-gramain.
